### PR TITLE
Refactors --gwas_cases (and removes --gwas_controls) to allow proportion of cases to be supplied by user

### PR DESCRIPTION
**Pull request by mmeier93**
# This PR does the following:

It adds the ability for the user to supply GWAS cases numbers (i.e. cases to simulate when simulating GWAS summary statistics) as a proportion of the total number of participants. This therefore removes the old mechanism in which the user had to supply both `--gwas_cases` and `--gwas_controls` as numbers.

# Details:

- Removed all trace of `--gwas_controls` in `main.nf` and `nextflow.config`.

- Updated documentation about `--gwas_cases` in main.nf and `docs/usage.md`, including how the pipeline handles the rounding of the number of cases when the proportion results in a float number.
 
- Removed the section under `Simulating GWAS summary statistics (using GCTA)` that was performing a check that the number of cases and controls to simulate match the total number of simulated participants (i.e. this was the section with `def cases_num`, `def controls_num`, `def total`).

- Inside Nextflow process called `simulate_gwas_sum_stats`, under `shell:`, added a section which defines `cases_num` and then rounds the `cases_num` by converting it to an integer. Then, `controls_num` is deduced from `rounded_case_num` and `total_participants`.

- In Nextflow processes `simulate_gwas_sum_stats` and `simulate_gen_and_sample`, explicitly added `params.num_participants` as an input, with `val`. Similarly, for `simulate_gwas_sum_stats`, also explicitly added `params.gwas_cases` as  an input with `val`. Indeed, not having it seemed to be causing issues with `-resume` not working (i.e. the cache would be kept even when changing the value of `params.gwas_cases` on the command line).

- Updated CI testing for the pipeline. 
   - Updated both `conf/cb.config` and `conf/test.config` to remove `--gwas_controls`.
   - For `test.config` added a matrix with different values was added in `ci.yml`.

# Comments:

- The way the matrix for `gwas_cases` is set in `ci.yml` could potentially be optimized.
- Testing of different proportion with `--gwas_cases` revealed that `GTCA` will give an error if the number of cases/controls in under 10. Added an issue: https://github.com/lifebit-ai/simulate/issues/49. Some checks could potentially be added to avoid this ever happening.

# Testing

- Lifebit CloudOS:
Successful: https://cloudos.lifebit.ai/app/jobs/5ff8296057851f011298acf3

- CI testing:
Successful: https://github.com/lifebit-ai/simulate/actions/runs/471103213

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ jobs:
     strategy:
       matrix:
         nxf_ver: ['20.01.0', '']
+        gwas_cases: ['0.40', '0.4', '0.41','0.423','0.47']
```
Comment (cgpu) [inline .github/workflows/ci.yml:11]: Only a reservation on naming, maybe worth swapping to more intuitive/ self-explanatory or `gwas_cases_proportion` even `cases_proportion`?

File: conf/test.config [hunk 2]
```diff
@@ -15,8 +15,6 @@ params  {
     simulate_vcf = true
     simulate_plink = true
     simulate_gwas_sum_stats = true
```
Comment (cgpu) [inline conf/test.config:17]: Here both `gwas_cases` and `gwas_controls` are removed correctly but we don't explicitly add the new replacer variable named  ~`cases_proportion`. Do we need to add this param in the test.config?

File: docs/usage.md [hunk 3]
```diff
@@ -41,9 +41,7 @@ nextflow run main.nf --num_participants 40
...
+- **--gwas_cases**: the number of cases to simulate for the GWAS summary statistics, represented as a fraction of --num_participants bigger than 0 and up to 1 (default: 0.5). Notably, if the proportion of cases results in a a float number of cases (i.e. for example if num_participants=40 and gwas_cases=0.42, then the number of cases would be 0.42*40 = 16.8), the pipeline will round down the number (i.e. in the example given, the number of cases will be set to 16). In addition, it should be noted that the final number of simulated cases may be smaller than the number provided by the user, due to how `GCTA` samples individuals for the simulation (i.e. it selects a subset of individuals in the provided genotype data). See more details at: https://cnsgenomics.com/software/gcta/#GWASSimulation.
```
Comment (cgpu) [inline docs/usage.md:44]: Naming of variable here, same as above to denote proportion is needed.

**Review by cgpu — APPROVED** [review]
Looks good!

Added a minor note only for the naming of the variable to indicate it is about fraction, but good to go

**Comment by mmeier93** [comment]
> Looks good!
> 
> Added a minor note only for the naming of the variable to indicate it is about fraction, but good to go

Thanks! 👍 Added the change to be made in the list of changes for the next iteration 👍

File: docs/usage.md (hunk 3 above)
Comment (mmeier93) [reply docs/usage.md:44]: I added the change to be made in the list of changes for the next iteration 👍

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mmeier93) [reply .github/workflows/ci.yml:11]: I added the change to be made in the list of changes for the next iteration 👍

**Pull request merged**